little edits on review of the new sd jwt encoding (#182)
oauth-wg/oauth-selective-disclosure-jwt: draft-ietf-oauth-selective-disclosure-jwt.md

SD-JWT = SD-JWT-DOC | SIG(SD-JWT-DOC, ISSUER-PRIV-KEY)
```

> `SD-CLAIMS` is an array of digest values that and ensure the integrity of and map to the respective Disclosures.  Digest values are calculated over the Disclosures, each of which contains the claim name (`CLAIM-NAME`) and value (`CLAIM-VALUE`) with a random salt (`SALT`). Digests are calculated using a digest derivation function such as a hash function, HMAC, or other (`DIGEST-DERIVATION()`):

Reviewer: ```suggestion
`SD-CLAIMS` is an array of digest values that ensures the integrity of and maps to the respective Disclosures.  Digest values are calculated over the Disclosures, each of which contains the claim name (`CLAIM-NAME`), the claim value (`CLAIM-VALUE`), and a random salt (`SALT`). Digests are calculated using a digest derivation function such as a hash function, HMAC, or other (`DIGEST-DERIVATION()`):
```

Reviewer 2: Best of both?
```suggestion
`SD-CLAIMS` is an array of digest values that ensure the integrity of and map to the respective Disclosures.  Digest values are calculated over the Disclosures, each of which contains the claim name (`CLAIM-NAME`), the claim value (`CLAIM-VALUE`), and a random salt (`SALT`). Digests are calculated using a digest derivation function such as a hash function, HMAC, or other (`DIGEST-DERIVATION()`):
```

`SD-CLAIMS` can also be nested deeper to capture more complex objects, as will be shown later.

> `SD-JWT` is sent from the Issuer to the Holder, together with the mapping of the plain-text claim values, the salt values, and potentially some other information.

Reviewer: Was this perceived redundant?

Author: Yeah, it felt redundant and somewhat out of order with the content just below.

Selectively disclosable claims are omitted from the SD-JWT. Instead, the hash digests of the respective Disclosures and potentially decoy digests are contained as an array in a new JWT claim, `_sd`.

The `_sd` claim MUST be an array of strings, each string being a hash digest of a Disclosure or a decoy digest as described above.
The array MAY be empty in case the Issuer decided not to selectively disclose any of the claims at that level. However, it is RECOMMENDED to omit `_sd` claim in this case to save space.

The Issuer MUST hide the original order of the claims in the array. To ensure this, it is RECOMMENDED to shuffle the array of hashes, e.g., by sorting it alphanumerically or randomly. The precise method does not matter as long as it does not depend on the original order of elements.

> Issuers MUST NOT issue SD-JWTs where:

Reviewer 2: I did not put a colon here as the introduction to the list is not a complete sentence.

```suggestion
Issuers MUST NOT issue SD-JWTs where
```

## Entropy of the salt {#salt-entropy}

> The security model that conceals the cleartext claims relies on the fact that the salt cannot be learned or guessed by

Reviewer: ```suggestion
The security model that conceals the plaintext claims relies on the fact that the salt cannot be learned or guessed by
```

Author: The draft currently uses plaintext, cleartext and plain text. Maybe a full reconciliation is needed at some point. In the meantime, I'll take this suggestion.
